sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1261,6 +1397,138 @@ no queue state is changed. This design is because the queues are already in
 the state the user desires, so if the function threw an exception in this case,
 the application would likely swallow it and then proceed.
 
+=== External events
+
+The default behavior is that the event associated with a graph node cannot be 
+used to synchronize workload outside the graph.
+To overcome this limitation, users can define and use graph exernal events.
+These events can be used as regular sycl events. Users can therefore use them
+to manage the synchronizations between different graphs running concurrently 
+or between the graph's kernels and regular operations submit to a sycl queue.
+If a graph containing one or more external events is run multiple times, 
+the external events are automatically reset when the graph is resubmitted to 
+the queue.
+
+[source,c++]
+----
+/* Basic usage example using the Record&Reply API. */
+
+// Create the graph `A`.
+sycl::queue queueA;
+sycl_ext::command_graph graphA(queueA.get_context(), queueA.get_device());
+
+// Start recording the queue.
+graphA.begin_recording(queueA);
+
+// Submit kernel on the recorded queue.
+// EventA is a graph-limited event that can only be used for managing
+// graph internal dependencies.
+EventA = queueA.submit([&](handler& cgh) {...});
+
+// Stop recording the queue.
+graphA.end_recording();
+
+event externalEventA;
+queueA.ext_oneapi_external_event(EventA, externalEventA, flase /* non-updatable */);
```

Comment:
I imagine another way to do this would be to pass a property to the `submit` call which requires the event to always be eternal, even when produced by a recording queue. Though I guess the reason to avoid this kind of design is that it would require modifying existing SYCL code to be used in record/replay?